### PR TITLE
[EDU] Write out multiple spool files, one per region, over SFTP

### PR DESCRIPTION
**Pull request by markolson**
* Obviously SFTP credentials will need to be put into the environment variables once we have them
* This should use a local file instead of SFTP when in development
* I opted to put the `EducationFacility` in the same namespace as the job itself since they're very tightly bound.

### CODE REVIEW

File: app/jobs/education_form/create_daily_spool_files.rb [hunk 1]
```diff
@@ -9,38 +11,65 @@ class CreateDailySpoolFiles
...
 
+    def run(day = Date.yesterday)
+      # Fetch all the records for the day
       records = EducationBenefitsClaim.unprocessed_for(day)
+      # Group the formatted records into different regions
+      regional_data = group_submissions(records)
+      # Create a remote file for each region, and write the records into them
+      create_files(day, regional_data)
+      # mark the records as processed
+      records.each { |r| r.update(processed_at: Time.zone.utc) }
```
Comment (austin-martinez) [inline app/jobs/education_form/create_daily_spool_files.rb]: Is this updating the data in the DB? If so, can be done in a single SQL statement with `update_all`

File: app/jobs/education_form/create_daily_spool_files.rb [hunk 2]
```diff
@@ -9,38 +11,65 @@ class CreateDailySpoolFiles
...
+      # TODO: Log the success/failure of the submission somewhere
+      true
+    end
 
+    def group_submissions(records)
```
Comment (austin-martinez) [inline app/jobs/education_form/create_daily_spool_files.rb]: `group_submissions_by_region` would be a more descriptive name here

File: app/jobs/education_form/education_facility.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+module EducationForm
+  class EducationFacility
+    # sourced from http://www.vba.va.gov/pubs/forms/VBA-22-1990-ARE.pdf
+
+    @default = :eastern
```
Comment (austin-martinez) [inline app/jobs/education_form/education_facility.rb]: Should this just be a constant `DEFAULT`?

File: app/jobs/education_form/create_daily_spool_files.rb [hunk 4]
```diff
@@ -9,38 +11,65 @@ class CreateDailySpoolFiles
...
+      regional_data = {}
```
Comment (austin-martinez) [inline app/jobs/education_form/create_daily_spool_files.rb]: Could remove L34 by using `Hash.new { |h, k| h[k] = [] }` here

File: config/education_benefits_regions.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+central:
```
Comment (austin-martinez) [inline config/education_benefits_regions.yml]: I don't see this being used anywhere

File: app/jobs/education_form/create_daily_spool_files.rb [hunk 6]
```diff
@@ -9,38 +11,65 @@ class CreateDailySpoolFiles
 
     WINDOWS_NOTEPAD_LINEBREAK = "\r\n"
 
-    def run(day = Date.yesterday)
-      # TODO: add and use the mapping of schools/regions -> spool file
+    attr_accessor :files
```
Comment (lihanli) [inline app/jobs/education_form/create_daily_spool_files.rb]: is this still needed?

Comment (markolson) [inline app/jobs/education_form/create_daily_spool_files.rb]: Nope, good catch

File: config/education_benefits_regions.yml (hunk 5 above)
Comment (markolson) [inline config/education_benefits_regions.yml]: Correct! Removed.

File: app/jobs/education_form/create_daily_spool_files.rb (hunk 4 above)
Comment (markolson) [inline app/jobs/education_form/create_daily_spool_files.rb]: I always forget you can set the default values. Thanks

File: app/jobs/education_form/create_daily_spool_files.rb (hunk 2 above)
Comment (markolson) [inline app/jobs/education_form/create_daily_spool_files.rb]: 👍

File: app/jobs/education_form/create_daily_spool_files.rb (hunk 1 above)
Comment (markolson) [inline app/jobs/education_form/create_daily_spool_files.rb]: 👍 I think I used `update_all` wrong at first somehow and convinced myself you couldn't use it on AR resultsets.

**Comment by lihanli** [comment]
:+1:

File: app/jobs/education_form/create_daily_spool_files.rb [hunk 7]
```diff
@@ -16,38 +18,58 @@ class CreateDailySpoolFiles < ActiveJob::Base
...
+    end
 
-      # TODO: SFTP the generated file(s)
-      # TODO: Mark the applications as processed once the send is successful
-      # TODO: Log the success/failure of the submission somewhere
-      true
+    def create_files(structured_data)
+      # TODO: Will be implemented in a follow-up PR.
+      Net::SFTP.start('host', 'username', password: 'password') do |sftp|
+        structured_data.each_with_object({}) do |(region, records), _localfiles|
```
Comment (austin-martinez) [inline app/jobs/education_form/create_daily_spool_files.rb]: Will `_localfiles` ever be used? Looking at this, you could just iterate over the hash with `.each` and not `.each_with_object`

**Comment by lihanli** [comment]
new changes LGTM

**Pull request merged**